vars/neoDeploy.groovy

```diff
@@ -19,33 +19,104 @@ def call(parameters = [:]) {
+            }
+        }
+
         def defaultDeployHost = script.commonPipelineEnvironment.getConfigProperty('DEPLOY_HOST')
         def defaultDeployAccount = script.commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
         def defaultCredentialsId = script.commonPipelineEnvironment.getConfigProperty('neoCredentialsId')
         if (defaultCredentialsId == null) {
             defaultCredentialsId = 'CI_CREDENTIALS_ID'
         }
 
-        def deployHost = utils.getMandatoryParameter(parameters, 'deployHost', defaultDeployHost)
-        def deployAccount = utils.getMandatoryParameter(parameters, 'deployAccount', defaultDeployAccount)
+        def deployHost
+        def deployAccount
+
+        if (deployMode.equals('MTA') || deployMode.equals('WAR_PARAMS')) {
+            deployHost = utils.getMandatoryParameter(parameters, 'deployHost', defaultDeployHost)
+            deployAccount = utils.getMandatoryParameter(parameters, 'deployAccount', defaultDeployAccount)
+        }
+
         def credentialsId = parameters.get('neoCredentialsId', defaultCredentialsId)
 
         def neoExecutable = getNeoExecutable(parameters)
 
         withCredentials([usernamePassword(
-                credentialsId: credentialsId,
-                passwordVariable: 'password',
-                usernameVariable: 'username'
-        )]) {
-            sh """#!/bin/bash
-                    "${neoExecutable}" deploy-mta \
-                      --user '${username}' \
+            credentialsId: credentialsId,
+            passwordVariable: 'password',
+            usernameVariable: 'username')]) {
+
+            def commonDeployParams =
+                """--user '${username}' \
+                   --password '${password}' \
+                   --source "${archivePath.getAbsolutePath()}" \
+                """
+
+            if (deployMode == 'MTA') {
+                sh """#!/bin/bash
+                      "${neoExecutable}" deploy-mta \
+                      ${commonDeployParams} \
                       --host '${deployHost}' \
-                      --source "${archivePath.getAbsolutePath()}" \
                       --account '${deployAccount}' \
-                      --password '${password}' \
                       --synchronous
-               """
+                   """
+            }
+
+            if (deployMode == 'WAR_PARAMS') {
+                sh """#!/bin/bash
+                      "${neoExecutable}" ${warAction} \
+                      ${commonDeployParams} \
+                      --host '${deployHost}' \
+                      --account '${deployAccount}' \
+                      --application '${applicationName}' \
+                      --runtime '${runtime}' \
+                      --runtime-version '${runtimeVersion}' \
+                      --size '${vmSize}'
+                   """
+            }
+
+            if (deployMode == 'WAR_PROPERTIESFILE') {
```

Comment:
Could you please convert the parameter values to lower case. Otherwise it would be different to other steps.